exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -270,4 +271,10 @@ public void setResultSet(AvaticaResultSet resultSet) {
   public void setUpdateCount(int value) {
     updateCount = value;
   }
+
+  @Override
+  public void setLargeMaxRows(long maxRowCount) throws SQLException {
+    execute("ALTER SESSION SET `" + ExecConstants.QUERY_MAX_ROWS + "`="+maxRowCount);
+    this.maxRowCount = maxRowCount;
```

Comment:
Looks like `maxRowCount` is taken from avatica. Does it provide similar functionality we want to implement?

Reply:
We need this here to ensure that when `getLargeMaxRows()` is called, we are reading it back from the value that was set using `setLargeMaxRows()`. Avatica only holds the value that has been set.